data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/configuration/SortingConfiguration.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.opensearch.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+public class SortingConfiguration {
+        private static final String DEFAULT_SORT_ORDER = "asc";
+
+        @NotEmpty
+        @NotNull
+        @JsonProperty("sort_key")
+        private String sortKey;
+        @JsonProperty("order")
```

Comment:
We will still need some validations for some things, such as `include` and `exclude` regex patterns (which we should compile on startup), and for order either being `asc` or `desc`. That can be handled in a future PR though